tests/datavolume_test.go

```diff
@@ -79,7 +79,33 @@ var _ = Describe("[Serial]DataVolume Integration", func() {
 	}
 
 	Describe("[rfe_id:3188][crit:high][vendor:cnv-qe@redhat.com][level:system] Starting a VirtualMachineInstance with a DataVolume as a volume source", func() {
-		Context("using Alpine import", func() {
+
+		Context("using DV with Alpine import", func() {
+			BeforeEach(func() {
+				By("Enable featuregate=HonorWaitForFirstConsumer")
+				// in practice we might have HonorWaitForFirstConsumer twice in featureGates array, but this is not a problem
+				jsonpath := `-o=jsonpath="{.spec.featureGates}"`
+				out, _, err := tests.RunCommand("kubectl", "get", "cdiconfigs.cdi.kubevirt.io", "config", jsonpath)
+				Expect(err).ToNot(HaveOccurred())
+
+				// no feature Gates? we need to add the whole structure, else just add the flag
+				// Looks like the output might be quoted
+				if str, err := strconv.Unquote(strings.TrimSpace(out)); str == "" {
+					patch := `{"spec":{"featureGates":["HonorWaitForFirstConsumer"]}}`
+					_, _, err = tests.RunCommand("kubectl", "patch", "cdiconfigs.cdi.kubevirt.io", "config", "-o=json", "--type=merge", "-p", patch)
+					Expect(err).ToNot(HaveOccurred())
+				} else {
+					patch := `[{"op": "add" , "path": "/spec/featureGates/0", "value": "HonorWaitForFirstConsumer"}]`
+					_, _, err = tests.RunCommand("kubectl", "patch", "cdiconfigs.cdi.kubevirt.io", "config", "--type=json", "-p", patch)
+					Expect(err).ToNot(HaveOccurred())
+				}
+			})
+			AfterEach(func() {
+				By("Restore featuregates")
+				patch := `[{"op": "remove" , "path": "/spec/featureGates/0", "value": "HonorWaitForFirstConsumer"}]`
+				_, _, err = tests.RunCommand("kubectl", "patch", "cdiconfigs.cdi.kubevirt.io", "config", "--type=json", "-p", patch)
+				Expect(err).ToNot(HaveOccurred())
+			})
 			It("[test_id:3189]should be successfully started and stopped multiple times", func() {
 
 				dataVolume := tests.NewRandomDataVolumeWithHttpImport(tests.GetUrl(tests.AlpineHttpUrl), tests.NamespaceTestDefault, k8sv1.ReadWriteOnce)
```

Comment:
do we have a functional test that exercises the flow of when a VMI references a DV by PVC when wffc is used?

Reply:
looks like there is none, and this one can be easily added, buy converting current test to table, and changing VMI definition

Reply:
or  better - I can just enable WFFC for the 5252 test